packages/opentelemetry-context-zone-peer-dep/src/util.ts

```diff
@@ -15,13 +15,17 @@
  */
 
 /**
- * check if an object has addEventListener and removeEventListener functions then it will return true.
- * Generally only called with a `TargetWithEvents` but may be called with an unknown / any.
+ * check if an object has `addEventListener` and `removeEventListener` functions.
+ * Generally only called with a `TargetWithEvents` but may be called with an `unknown` value.
  * @param obj - The object to check.
  */
-export function isListenerObject(obj: any = {}): boolean {
+export function isListenerObject(obj: unknown): boolean {
   return (
+    typeof obj === 'object' &&
+    obj !== null &&
+    'addEventListener' in obj &&
     typeof obj.addEventListener === 'function' &&
+    'removeEventListener' in obj &&
     typeof obj.removeEventListener === 'function'
```

Comment:
There are alternatives without changing the runtime code (which can impose a small runtime cost), but this seemed the most straightforward/clear, and doesn't seem like the micro performance cost should matter all that much.